react/src/pages/analysis/Analysis.tsx

```diff
@@ -152,48 +198,92 @@ export default function Analysis() {
+                            <TextField
+                            multiline
+                            value={props.value}
+                            onChange={event => props.onChange(event.target.value)}
+                            rows={4}
+                            fullWidth
+                            />
+                        )}
                     ]}
-                    data={analyses}
+                    data={rows}
                     title={
                         <Title>Active Analyses</Title>
                     }
                     options={{
                         pageSize: 10,
                         filtering: true,
-                        search: false
+                        search: false,
+                        padding: 'dense'
                     }}
                     actions={[
                         {
-                            icon: VisibilityIcon,
+                            icon: Visibility,
                             tooltip: 'Analysis details',
                             onClick: (event, rowData) => {
-                                setActiveRow((rowData as AnalysisRun))
+                                setActiveRow((rowData as AnalysisRow))
                                 setDetail(true)
                             }
                         },
                         rowData => ({
-                            icon: CancelIcon,
+                            icon: Cancel,
                             tooltip: 'Cancel analysis',
                             onClick: () => {
                                 setActiveRow(rowData)
                                 setCancel(true)
                             },
-                            disabled: rowData.status !== PipelineStatus.RUNNING,
+                            disabled: rowData.state !== PipelineStatus.RUNNING,
                         }),
                         {
-                            icon: AddIcon,
+                            icon: Add,
                             tooltip: 'Add New Analysis',
                             isFreeAction: true,
                             onClick: (event) => setDirect(true)
                         }
                     ]}
+                    editable={{
+                        onRowUpdate: (newData, oldData) =>
+                            new Promise((resolve, reject) => {
+                                const dataUpdate = [...rows];
+                                // find the row; assume analysis_id is unique
+                                const index = dataUpdate.findIndex((row, index, obj) => {
+                                    return row.analysis_id === oldData?.analysis_id
+                                });
+
+                                // only update the rows that are allowed
+                                dataUpdate[index].notes = newData.notes;
+                                dataUpdate[index].result_hpf_path = newData.result_hpf_path;
```

Comment:
This still mutates the previous state because this is not a deep copy, which might have unexpected consequences. I think you should update the properties on a copy of `dataUpdate[index]` and then replace the array entry, if that makes sense.